protocols/gossipsub/src/behaviour.rs

```diff
@@ -538,9 +539,10 @@ where
         }
 
         // send subscription request to all peers
-        for (peer_id, peer) in self.connected_peers.iter_mut() {
+        for peer_id in self.connected_peers.keys().copied().collect::<Vec<_>>() {
```

Comment:
Unfortunately it's needed because otherwise it's a immutable ref to self, which conflicts with the below call to `self.send_message`. It's also how it's currently implemented on master:

https://github.com/jxs/rust-libp2p/blob/3837e33cd4c40ae703138e6aed6f6c9d52928a80/protocols/gossipsub/src/behaviour.rs#L529-L534

Reply:
yeah, I tried to improve it but it needs quite some changes that I think makes sense to do on a follow up PR